src/test/scala/SimpleKafkaConsumer.scala

```diff
@@ -0,0 +1,49 @@
+import java.util.Properties
+
+import com.fasterxml.jackson.databind.KeyDeserializer
+import org.apache.kafka.clients.consumer.{ConsumerRecords, KafkaConsumer}
+import org.apache.kafka.common.serialization.Deserializer
+import net.manub.embeddedkafka.Codecs.stringDeserializer
+import net.manub.embeddedkafka.ConsumerExtensions._
+
+class SimpleKafkaConsumer[K,V](consumerProps : Properties,
+                               topic : String,
+                               keyDeserializer: Deserializer[K],
+                               valueDeserializer: Deserializer[V],
+                               function : ConsumerRecords[K, V] => Unit,
+                               poll : Long = 2000) {
+
+  private var running = false
+
+  private val consumer = new KafkaConsumer[K, V](consumerProps, keyDeserializer, valueDeserializer)
+
+
+  private val thread = new Thread {
+    import scala.collection.JavaConverters._
+
+    override def run: Unit = {
+      consumer.subscribe(List(topic).asJava)
+      consumer.partitionsFor(topic)
+
+      while (running) {
```

Comment:
The poll value that's passed into consumer.poll might be better named as timeout.

Check running
Execute poll, wait until max buffered records is reached or the specified timeout.
send records to user defined function
Check running
....